src/test/regress/sql/resync_metadata.sql

```diff
@@ -0,0 +1,48 @@
+CREATE SCHEMA resync_metadata;
+SET search_path TO resync_metadata;
```

Comment:
As we discussed this further, we've realized that we should break dependency btw sequence and table first, instead of passing the flag to worker_drop_distributed_table.

Reply:
done and added relevant tests. Also added `alter_distributed_table` test